lib/configure.js

```diff
@@ -144,17 +139,18 @@ function configure (gyp, argv, callback) {
     // disable -T "thin" static archives by default
     variables.standalone_static_library = gyp.opts.thin ? 0 : 1
 
-    if (vsSetup) {
-      // GYP doesn't (yet) have support for VS2017, so we force it to VS2015
-      // to avoid pulling a floating patch that has not landed upstream.
-      // Ref: https://chromium-review.googlesource.com/#/c/433540/
-      gyp.opts.msvs_version = '2015'
-      process.env['GYP_MSVS_VERSION'] = 2015
-      process.env['GYP_MSVS_OVERRIDE_PATH'] = vsSetup.path
-      defaults['msbuild_toolset'] = 'v141'
-      defaults['msvs_windows_target_platform_version'] = vsSetup.sdk
-      variables['msbuild_path'] = path.join(vsSetup.path, 'MSBuild', '15.0',
-                                            'Bin', 'MSBuild.exe')
+    if (win) {
+      process.env['GYP_MSVS_VERSION'] = Math.min(vsInfo.versionYear, 2015)
+      process.env['GYP_MSVS_OVERRIDE_PATH'] = vsInfo.path
+      defaults['msbuild_toolset'] = vsInfo.toolset
+      if (vsInfo.sdk) {
+        defaults['msvs_windows_target_platform_version'] = vsInfo.sdk
+      }
+      if (vsInfo.versionMajor > 15 ||
+          (vsInfo.versionMajor === 15 && vsInfo.versionMajor >= 9)) {
+        defaults['msvs_enable_marmasm'] = 1
```

Comment:
IMO this is only needed if `variables.target_arch == 'arm64'`

Reply:
Added a fixup, thanks @refack!